dev/com.ibm.ws.org.apache.cxf.cxf.core.3.2/src/org/apache/cxf/message/MessageImpl.java

```diff
@@ -1385,5 +1388,24 @@ public Object merge(String key, Object value, BiFunction<? super Object, ? super
     public String[] getPropertyNames() {
         return propertyNames;
     }
+
+    @FFDCIgnore(ConcurrentModificationException.class)
+    @Override
+    public String toString() {
+        StringBuilder sb = new StringBuilder();
+        sb.append('{');
+        for (int i=0; i<TOTAL; i++) {
+            if (propertyValues[i] != NOT_FOUND) {
+                sb.append(propertyNames[i]).append("=").append(propertyValues[i]).append(", ");
+            }
+        }
+        try {
+            forEach((k, v) -> { sb.append(k).append("=").append(v).append(", "); });
```

Comment:
The last name/value isn't going to look pretty with a trailing ", ", but not a big deal. I'll let you decide if you want to resolve that.

Reply:
yeah, I'm fine with a trailing comma - we should only ever see it when trace is enabled - and we'll only be looking at a trace if something uglier has already happened.  :)